fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java

```diff
@@ -302,14 +258,10 @@ private void refreshTable(BaseTable updatedTable, long baseSnapshotId,
         IcebergTableName updatedIcebergTableName = new IcebergTableName(dbName, tableName, updatedSnapshotId);
         long latestRefreshTime = tableLatestRefreshTime.computeIfAbsent(new IcebergTableName(dbName, tableName), ignore -> -1L);
 
-        List<String> updatedPartitionNames = updatedTable.spec().isPartitioned() ?
-                listPartitionNamesWithSnapshotId(updatedTable, dbName, tableName, updatedSnapshotId, executorService) :
-                new ArrayList<>();
-
+        partitionCache.invalidate(baseIcebergTableName);
+        partitionCache.getUnchecked(updatedIcebergTableName);
         synchronized (this) {
-            partitionNames.put(updatedIcebergTableName, updatedPartitionNames);
```

Comment:
table name contains snapshotId. so you should drop the cache value of table name with old snapshotId.

Reply:
```
partitionCache.invalidate(baseIcebergTableName);
        partitionCache.getUnchecked(updatedIcebergTableName);
        synchronized (this) {
            tables.put(updatedIcebergTableName, updatedTable);
        }
```

Does it work?